lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -26,7 +26,7 @@ def __init__(self, context):
     def execute(self) -> int:
         errors = 0
 
-        groups = self.context.group.split(",")
+        groups = [group for group in self.context.group.split(",") if group]
```

Comment:
I dont think we had enough discussion around it. AFAIR, we thought that even if someone provides this `--group=group1,,,group2` we shouldn't break out of the loop if we encounter an empty value since there could present other groups to clear and that's why I introduced this change. Now alternatively, we could actually loop over with empty values and then log that we encounter an empty value, and make `error` value as 1 but then continue. Or do you mean we should just abort our operation when we encounter an empty string?

Reply:
I remember we discussed how easy it would be to get this with a `--tools=${tool1},${tool2},${tool2}` situation ... I don't think that necessarily provides a clear prescription. Is it a user error that they're referencing undefined environment variables here, or should we be friendly by ignoring them?

I don't see an unambiguous "right" answer. Sure, it's easy for them to avoid this situation, so "failing" doesn't present an insurmountable obstacle for the user; and I suppose we might argue that ignoring the blank values presents a risk of proceeding with unintended input.

I did think about that latter clause when I saw the new code, and decided I didn't feel sufficiently strongly about it to complain. Webb, are you saying you _do_ care, or are you just idly kicking the ant hill? 😆

Reply:
If we continue, then it's a WARNING, not an ERROR; that is, you didn't do it wrong, and everything will work, but it might be better to do it differently in the future. I don't feel strongly that it's worth telling them about details like this if we're going to go ahead and do it anyway. I suppose we could word the warning gently, like "if you didn't intend to have empty tools/remotes/groups maybe you did something you didn't intend", but ultimately that sort of messaging can be more annoying than helpful.

Reply:
> "if you didn't intend to have empty tools/remotes/groups maybe you did something you didn't intend"

I would go with something more like "blank value specified for [group|remote|tool] ignored" -- I hope the part about unintended actions should be sufficiently obvious.